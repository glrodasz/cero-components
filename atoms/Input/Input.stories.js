import { Input, options, styles } from '.'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(Input, styles)
const ListTemplate = getListTemplate(Input, styles)

export default {
  title: 'Atoms/Input',
  component: Input,
  args: {
    placeholder: 'Kingdom slowed',
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
  },
}

export const Default = Template.bind({})

export const Value = Template.bind({})
Value.args = { value: 'Others mainly cakehole Thorin moldy facial.' }

export const Types = ListTemplate.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }

export const Inline = Template.bind({})
Inline.args = { isInline: true }
