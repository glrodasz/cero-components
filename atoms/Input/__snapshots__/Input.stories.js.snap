// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Input Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</div>
`;

exports[`Storyshots Atoms/Input Inline 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    className="input is-inline"
    onChange={[Function]}
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</div>
`;

exports[`Storyshots Atoms/Input Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Kingdom slowed"
    type="text"
    value=""
  />
</div>
`;

exports[`Storyshots Atoms/Input Value 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Kingdom slowed"
    type="text"
    value="Others mainly cakehole Thorin moldy facial."
  />
</div>
`;
